test/compile/data/bin.test.ts

```diff
@@ -45,110 +43,59 @@ describe('compile/data/bin', function() {
-            field: 'Rotten_Tomatoes_Rating',
-            type: 'quantitative'
-          },
-          size: {
-            aggregate: 'count',
-            field: '*',
-            type: 'q'
+            bin: {"maxbins": 10},
+            field: "Rotten_Tomatoes_Rating",
+            type: "q"
           }
         }
       });
-      const transformArray = bin.parseUnit(model);
-      const transformArrayVals = vals(transformArray);
-      const imdbTransform = transformArrayVals[0];
-      const rotten10Transform = transformArrayVals[1];
-      const rotten20Transform = transformArrayVals[2];
-      it('should be binned twice', function() {
-        assert.deepEqual(imdbTransform.length, 2);
-        assert.deepEqual(imdbTransform[0], {
-          type: 'extent',
-          field: 'IMDB_Rating',
-          signal: 'bin_maxbins_10_IMDB_Rating_extent'
-        });
-        assert.deepEqual(imdbTransform[1],{
-          type: 'bin',
-          field: 'IMDB_Rating',
-          as: [ 'bin_maxbins_10_IMDB_Rating_start', 'bin_maxbins_10_IMDB_Rating_end' ],
-          signal: 'bin_maxbins_10_IMDB_Rating_bins',
-          maxbins: 10,
-          extent: {signal: 'bin_maxbins_10_IMDB_Rating_extent'}
-        });
-
-        assert.deepEqual(imdbTransform.length, 2);
-        assert.deepEqual(rotten10Transform[0], {
-          type: 'extent',
-          field: 'Rotten_Tomatoes_Rating',
-          signal: 'bin_maxbins_10_Rotten_Tomatoes_Rating_extent'
-        });
-        assert.deepEqual(rotten10Transform[1], {
-          type: 'bin',
-          field: 'Rotten_Tomatoes_Rating',
-          as:
-          [ 'bin_maxbins_10_Rotten_Tomatoes_Rating_start',
-            'bin_maxbins_10_Rotten_Tomatoes_Rating_end' ],
-          signal: 'bin_maxbins_10_Rotten_Tomatoes_Rating_bins',
-          maxbins: 10,
-          extent: {signal: 'bin_maxbins_10_Rotten_Tomatoes_Rating_extent'}
-        });
-
-        assert.deepEqual(rotten20Transform.length, 3);
-        assert.deepEqual(rotten20Transform[0], {
-          type: 'extent',
-          field: 'Rotten_Tomatoes_Rating',
-          signal: 'bin_maxbins_20_Rotten_Tomatoes_Rating_extent'
-        });
-        assert.deepEqual(rotten20Transform[1], {
-          type: 'bin',
-          field: 'Rotten_Tomatoes_Rating',
-          as:
-          [ 'bin_maxbins_20_Rotten_Tomatoes_Rating_start',
-            'bin_maxbins_20_Rotten_Tomatoes_Rating_end' ],
-          signal: 'bin_maxbins_20_Rotten_Tomatoes_Rating_bins',
-          maxbins: 20,
-          extent: {signal: 'bin_maxbins_20_Rotten_Tomatoes_Rating_extent'}
-        });
-        assert.deepEqual(rotten20Transform[2], {
-          type: 'formula',
-          as: 'bin_maxbins_20_Rotten_Tomatoes_Rating_range',
-          expr: 'format(datum["bin_maxbins_20_Rotten_Tomatoes_Rating_start"], \'s\') + \' - \' + format(datum["bin_maxbins_20_Rotten_Tomatoes_Rating_end"], \'s\')'
-        });
+      const transform = vals(bin.parseUnit(model))[0];
```

Comment:
The test right now doesn't really check if it apply the bin only once.  Make sure to assert the length...